Add CONTRIBUTING.md (#34)
storm-devs/storm-engine: CONTRIBUTING.md

# Contributing

When contributing to this repository, please check [Issues](https://github.com/storm-devs/storm-engine/issues) and
[Projects](https://github.com/storm-devs/storm-engine/projects),

> but you will probably start a [discussion](https://github.com/storm-devs/storm-engine/discussions) first.

Reviewer: I'd also mention the discord channel here:

> When contributing to this repository, please check [Issues](https://github.com/storm-devs/storm-engine/issues) and
[Projects](https://github.com/storm-devs/storm-engine/projects). Also, it's a good idea to drop a message on our [Discord channel](https://discord.gg/jmwETPGFRe) or start a [discussion](https://github.com/storm-devs/storm-engine/discussions) first to let others know what you're planning to work on.

Reviewer 2: I think we should leave a link to contributing.md in that section

## Styleguides

> ### Git commit messages

Reviewer 3: capitalization
```suggestion
### Git Commit Messages
```

* Use the present tense and imperative mood
* Limit the first line to 72 characters

> ### Code style

Reviewer 3: ```suggestion
### Code Style
```

To keep the source consistent and easy to merge, we are using ClangFormat to format code.
You can find ``.clang-format`` config in the root.

## Pull Request Process

> To make approve process easier, pull request, if applicable, should follow this:

Reviewer 3: I'd rephrase: 
```suggestion
To make the approval process easier and faster, your pull request should include the following (if applicable):
```

* A link to issue or project card
* Reasoning of changes
* Summary of changes

There are also some tips:

* If you are working on different features, split the job to branches
* Prefer to rebase over merge when pulling changes from master
* Avoid mixing tasks in a single Pull Request
* Give understandable names for commits

> * Be sure that all your change is clear to understand, add have comments for complicated code

Reviewer 3: ```suggestion
* Be sure that all your changes are clear to understand.
* Add comments for non-trivial code
```
